### PR TITLE
Service Accounts: Fix issue in UserTable and ServiceAccountsTable

### PR DESCRIPTION
**Pull request by kalleep**
**What this PR does / why we need it**:
In https://github.com/grafana/grafana/pull/43573 changes to how showRemoveModal property is set in UserTable causes a bug where a confirmation modal is rendered for every item in the list with the last one on top.

This happens because the ConfirmModal is rendered in a loop and if we use true as value then every item will render the modal.

Changes to UserTable - https://github.com/grafana/grafana/pull/43885

### CODE REVIEW

**Review by eleijonmarck — APPROVED** [review]

**Comment by eleijonmarck** [comment]
this probably fixes the bug. But we should probably fix the issue "more" permanent" or true to make it use the boolean

**Comment by eleijonmarck** [comment]
@dprokop should have a look at this

**Comment by kalleep** [comment]
> this probably fixes the bug. But we should probably fix the issue "more" permanent" or true to make it use the boolean

Yes, a better solution would be to move out the ConfirmModal and fix the state. I can look into it tomorrow :)

**Comment by dprokop** [comment]
@kalleep @eleijonmarck  - created a PR with the alternative solution to this: https://github.com/grafana/grafana/pull/43885

It takes the ConfirmModal out of the context of a single user row. I have not updated the ServiceAccountsTable tho, so please do so. There is no boolean/string state anymore. The rendering relies on the `userToRemove: OrgUser | null` being present in the state.

**Review by dprokop — CHANGES_REQUESTED** [review]
@kalleep @eleijonmarck - created a PR with the alternative solution to this: #43885

It takes the ConfirmModal out of the context of a single user row. I have not updated the ServiceAccountsTable tho, so please do so. There is no boolean/string state anymore. The rendering relies on the userToRemove: OrgUser | null being present in the state.

**Comment by eleijonmarck** [comment]
@dprokop great with the tests!

File: public/app/features/serviceaccounts/ServiceAccountsTable.tsx [hunk 1]
```diff
@@ -43,91 +44,92 @@ const ServiceaccountsTable: FC<Props> = (props) => {
...
+                  </td>
+                )}
+              </tr>
+            );
+          })}
+        </tbody>
+      </table>
+      {toRemove !== null && (
+        <ConfirmModal
+          body={`Are you sure you want to delete serviceaccount ${toRemove.login}?`}
```
Comment (dprokop) [inline public/app/features/serviceaccounts/ServiceAccountsTable.tsx:123]:
```suggestion
          body={`Are you sure you want to delete ${toRemove.login} service account?`}
```

**Review by dprokop — APPROVED** [review]

**Pull request merged**